apps/govern/components/Contract.tsx

```diff
@@ -0,0 +1,93 @@
+import { LoadingOutlined } from '@ant-design/icons';
+import { Card, Flex, Space, Spin, Typography } from 'antd';
+import { useContractParams } from 'hooks/index';
+import { useRouter } from 'next/router';
+import { useEnsName } from 'wagmi';
+
+import { useAppSelector } from 'store/index';
+import styled from 'styled-components';
+
+import { UNICODE_SYMBOLS } from 'common-util/constants/unicode';
+import { EXPLORE_URLS } from 'common-util/constants/urls';
+import { CHAIN_NAMES, getAddressFromBytes32, truncateAddress } from 'common-util/functions';
+
+const StyledMain = styled.main`
+  display: flex;
+  flex-direction: column;
+  max-width: 800px;
+  margin: 0 auto;
+`;
+
+const Title = styled.h1`
+  font-size: 24px;
+  margin: 0 0 24px;
+`;
+
+const { Text, Paragraph } = Typography;
+
+export const ContractPage = () => {
+  const { stakingContracts } = useAppSelector((state) => state.govern);
+  const router = useRouter();
+  const addressParam = router?.query?.address;
+
+  const contract = stakingContracts.find((item) => item.address === addressParam);
+
+  const formattedAddress = contract ? getAddressFromBytes32(contract.address) : '';
+
+  const { data: contractParams } = useContractParams(
+    formattedAddress,
+    contract?.chainId || 0,
+    !!contract,
+  );
+
+  const { data: ensName, isFetching: isEnsNameFetching } = useEnsName({
```

Comment:
this may fail on polygon as it does not have a universalEnsContract deployed

you may need to create a `publicClient` that only reads ethereum, `mainnet`, and use this as the client via viem instead.

unsure though, just a point to check.